pkg/base/replicated.go

```diff
@@ -152,7 +157,149 @@ func renderReplicated(u *upstreamtypes.Upstream, renderOptions *RenderOptions) (
 		}
 	}
 
-	return &commonBase, helmBases, nil
+	return &commonBase, helmBases, kotsKindsFileMap, nil
+}
+
+func renderKotsKinds(upstreamFiles []upstreamtypes.UpstreamFile, renderedConfig *kotsv1beta1.Config, renderOptions *RenderOptions, builder *template.Builder) (*kotsutil.KotsKinds, map[string][]byte, error) {
+	fileContentsMap := make(map[string][]byte)
+	kotsKinds := kotsutil.EmptyKotsKinds()
+
+	for _, upstreamFile := range upstreamFiles {
+		document := &Document{}
+		if err := yaml.Unmarshal(upstreamFile.Content, document); err != nil {
+			continue
+		}
+
+		decode := scheme.Codecs.UniversalDeserializer().Decode
+		decoded, gvk, err := decode(upstreamFile.Content, nil, nil)
```

Comment:
instead of decoding, i would suggest utilizing [this function](https://github.com/replicatedhq/kots/blob/237fa185de3a4dde7fca15c4f0b6635b99a1bbda/pkg/base/base.go#L279) and `continue` if it's not a kots kind.